src/Drivers/Decorator.php

```diff
@@ -520,6 +610,28 @@ public function instance($name)
         return fn () => $feature;
     }
 
+    /**
+     * Retrieve the defined features for the given scope.
+     *
+     * @internal
+     *
+     * @param  mixed  $scope
+     * @return \Illuminate\Support\Collection<int, string>
+     */
+    public function definedFeaturesForScope($scope)
+    {
+        return collect($this->nameMap)
+            ->only($this->defined())
```

Comment:
@timacdonald Unfortunately this doesn't work with 3rd party services like LaunchDarkly where the features are defined in their platform.

Any thoughts on adding a check before this to see the `namedMap` is empty and if it is we load all of the features from `$this->defined()`?

An alternative could be that I retrieve all of the features from LaunchDarkly within a service provider and define them. But this just seems wrong as I can do the same thing within the `defined` method on the driver.

Reply:
What are your thoughts on having a contract the drivers can implement to provide a list of features for the given scope?

<img width="943" alt="Screenshot 2024-11-11 at 11 17 37" src="https://github.com/user-attachments/assets/5e68ef5b-2767-45bf-acbb-81e53f97cfc6">

I suppose the thing we are trying to solve in this PR also impacts 3rd party drivers as well, so feels like we need a way for them to handle things themselves.

Reply:
This approach would be perfect!

Reply:
Opened a PR: https://github.com/laravel/pennant/pull/127